src/pages/RequestView/RemakeRequestButton/RemakeRequestButton.tsx

```diff
@@ -0,0 +1,103 @@
+import { Button } from '@material-ui/core'
+import { JobRequestCreationContext } from 'components/JobRequestCreation'
+import { useCommandsParameterized } from 'hooks/useCommandParameterized'
+import {
+  buttonText,
+  CannotReExecuteButton,
+  DummyRequest,
+  systemCommandPairFromArgs,
+} from 'pages/RequestView/RemakeRequestButton/remakeRequestButtonHelpers'
+import { useContext } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { Request } from 'types/backend-types'
+import { SystemCommandPair } from 'types/custom-types'
+import {
+  CommandViewModelComment,
+  CommandViewModelParameters,
+  CommandViewRequestModel,
+} from 'types/form-model-types'
+
+interface RemakeRequestButtonProps {
+  request: Request | undefined
+}
+
+const RemakeRequestButton = ({ request }: RemakeRequestButtonProps) => {
+  const navigate = useNavigate()
+  const { setSystem, setCommand, setIsJob, setRequestModel, setIsReplay } =
+    useContext(JobRequestCreationContext)
+  const {
+    command: commandName,
+    system: systemName,
+    system_version: systemVersion,
+    namespace,
+    instance_name: instanceName,
+    parameters: theParameters,
+    comment: theComment,
+  } = request || DummyRequest
+  const { commands: systemCommandPair } =
+    useCommandsParameterized<SystemCommandPair>(
+      systemCommandPairFromArgs(
+        commandName,
+        systemName,
+        systemVersion,
+        namespace,
+      ),
+    )
+
+  if (
+    !setSystem ||
+    !setCommand ||
+    !setIsJob ||
+    !setRequestModel ||
+    !setIsReplay
+  ) {
+    return <CannotReExecuteButton message="ERROR: Unknown error" />
```

Comment:
It's not generally a bad idea for a refactor, but honestly I think in practice that would make it *more* complex rather than less.

Notice that in the `CannotReExecuteButton` there's more to it than just `disabled` and the tooltip. There is `component=div` and `pointerEvents: 'auto'` in the style. These are present because providing a child button that is disabled to a `Tooltip` component throws an error.

In trying a quick refactor, by setting flags and messages instead of returning, the guards that the `return <CannotReExecuteButton message="..." />` provide are eliminated. So then the calls to `setSystem`, etc. are broken and either an unsafe cast has to be done or some other solution has to be invented.

Moving on, the return would then have to be something like

```typescript
    return (
    <Tooltip title={buttonTooltipMessage} data-testid="reexecute-button">
      <Button
        component={canReExecute ? "button" : "div"}
        disabled={!canReExecute}
        variant="contained"
        color="primary"
        style={
            {...{ float: 'right'},
             ...{canReExecute ? null: {pointerEvents: 'auto' }}
        }
      >
        {'Remake Request'}
      </Button>
    </Tooltip>
  )
```
which to me (along with whatever kludgy workaround I'd come up with for `setSystem`, etc.) makes the whole thing harder to understand.

Maybe a more elegant refactor could be possible, but honestly I'm not sure the issue of complexity is so overbearing in this case to make it worth the effort.